generator/src/main/java/com/scottlogic/deg/generator/generation/ChecksummedCodeStringGenerator.java

```diff
@@ -32,30 +68,57 @@ public ChecksummedCodeStringGenerator(RegexStringGenerator generator, boolean ne
 
     @Override
     public StringGenerator intersect(StringGenerator stringGenerator) {
+        if (stringGenerator instanceof ChecksummedCodeStringGenerator) {
```

Comment:
This use of instanceof strikes me as an anti-pattern. For every new implementation of `StringGenerator` we will need to modify every existing implementation of `StringGenerator`.

Reply:
Yes, fair point.  It was there in the previous code, but its use has grown a bit here now.  What would you suggest as an alternative?